src/tools/simeng/main.cc

```diff
@@ -91,8 +91,9 @@ int main(int argc, char** argv) {
 
   // Create the instance of the core to be simulated
   std::unique_ptr<simeng::CoreInstance> coreInstance =
-      std::make_unique<simeng::CoreInstance>(OS.getSyscallHandler(), memory,
-                                             mmu);
+      std::make_unique<simeng::CoreInstance>(
+          memory, mmu,
+          [&](auto syscallInfo) { OS.receiveSyscall(syscallInfo); });
```

Comment:
Rather than creating callback here would it better to make SimOS return the callback (This way you'll also be able to typedef the callback making the intent more clear)
```
<typedef-return-type> SimOS::getSyscallReceiver() {
   return [this](auto SyscallInfo) {
    receiveSyscall(SyscallInfo);
    }
}
```

Reply:
We're going down of path of typedef'ing our functors so agree with this change